sdk/tests/js/webgl-test-utils.js

```diff
@@ -3136,38 +3128,30 @@ var runSteps = function(steps) {
  * @param {!function(!HTMLVideoElement): void} callback Function to call when
  *        video is ready.
  */
-function startPlayingAndWaitForVideo(video, callback) {
+async function startPlayingAndWaitForVideo(video, callback) {
   if (video.error) {
     testFailed('Video failed to load: ' + video.error);
     return;
   }
 
-  video.addEventListener(
-      'error', e => { testFailed('Video playback failed: ' + e.message); },
-      true);
-
-  var rvfc = getRequestVidFrameCallback();
-  if (rvfc === undefined) {
-    var timeWatcher = function() {
-      if (video.currentTime > 0) {
-        callback(video);
-      } else {
-        requestAnimFrame.call(window, timeWatcher);
-      }
-    };
-
-    timeWatcher();
-  } else {
-    // Calls video.requestVideoFrameCallback(_ => { callback(video) })
-    rvfc.call(video, _ => { callback(video) });
-  }
-
   video.loop = true;
   video.muted = true;
   // See whether setting the preload flag de-flakes video-related tests.
   video.preload = 'auto';
-  video.play();
-};
+
+  try {
+    await video.play();
+  } catch (e) {
+    testFailed('video.play failed: ' + e);
+    return;
+  }
+
+  if (video.requestVideoFrameCallback) {
```

Comment:
This is probably fine, but may slightly slow down test execution in Chromium. The old code is better for rVFC, since it may fire before play() is called. In an ideal world this function would set the rVFC callback then the `video.src` and never even bother calling `play()` since that's just a source of flakiness as far as the conformance pixel tests are concerned.

Without keeping the rest of the error checking though, it's hard to keep the rVFC path since the promise rejection is pulling double duty for error detection and play denial.

I think it's fine to land this and revert if we see issues (or improvements!), but defer to @kenrussell. If we want to avoid even minor test slow down risk, the rVFC path should set rVFC before `play()` and only `try { video.play(); } catch (e) {...}` without the `await`.

You might at least remove the `await new Promise(...)` in favor of: 

```
if (video.requestVideoFrameCallback) {
  video.requestVideoFrameCallback(_ => { callback(video); });
  return;
}
```

Reply:
Having a unified codepath that optionally awaits for the rvfc is nicer than divergent behavior. Also using rvfc to carry the actual callback means the function as a whole isn't awaitable anymore.

I can't imagine any serious perf differences between rvfc before play or not. (Really, I don't think rvfc should be spec-allowed to callback before await video.play does)

Thanks for the feedback!